applications/ShapeOptimizationApplication/python_scripts/optimizer_factory.py

```diff
@@ -42,16 +63,35 @@ def CreateOptimizer(optimization_settings, model, external_analyzer=EmptyAnalyze
 
     optimization_settings.ValidateAndAssignDefaults(default_settings)
 
-    model_part_controller = model_part_controller_factory.CreateController(optimization_settings["model_settings"], model)
-
-    analyzer = analyzer_factory.CreateAnalyzer(optimization_settings, model_part_controller, external_analyzer)
-
-    communicator = communicator_factory.CreateCommunicator(optimization_settings)
-
-    if optimization_settings["design_variables"]["type"].GetString() == "vertex_morphing":
-        return VertexMorphingMethod(optimization_settings, model_part_controller, analyzer, communicator)
-    else:
-        raise NameError("The following type of design variables is not supported by the optimizer: " + variable_type)
+# ------------------------------------------------------------------------------
+def ValidateObjectives(objectives):
```

Comment:
`objectives` -> `objective_parameters`

The term response, objective, constraint is used for many different types of objects, e.g. the parameters from the json, the dict in the communicator...

Reply:
You are right, I changed it here. At some point, we should also make this more consistent throughout the entire app.